virtcontainers/kata_agent.go

```diff
@@ -742,6 +743,34 @@ func (k *kataAgent) setProxyFromGrpc(proxy proxy, pid int, url string) {
 	k.state.URL = url
 }
 
+func (k *kataAgent) getDNS(sandbox *Sandbox) ([]string, error) {
+	ociSpecJSON, ok := sandbox.config.Annotations[vcAnnotations.ConfigJSONKey]
+	if !ok {
+		return nil, errorMissingOCISpec
+	}
+
+	ociSpec := &specs.Spec{}
+	if err := json.Unmarshal([]byte(ociSpecJSON), ociSpec); err != nil {
+		return nil, err
+	}
+
+	ociMounts := ociSpec.Mounts
+
+	for _, m := range ociMounts {
+		if m.Destination == GuestDNSFile {
+			content, err := ioutil.ReadFile(m.Source)
+			if err != nil {
+				return nil, fmt.Errorf("Could not read file %s", m.Source)
```

Comment:
You are discarding the error here - can you add it to the returned error as it will indicate *why* the file wasn't readable.

Reply:
Thanks @jodh-intel. Updated the PR.